include/eve/module/real/core/function/regular/simd/x86/rec.hpp

```diff
@@ -13,53 +13,102 @@
 
 namespace eve::detail
 {
-  // -----------------------------------------------------------------------------------------------
-  // 128 bits implementation
   template<floating_real_scalar_value T, typename N>
-  EVE_FORCEINLINE wide<T, N> rec_(EVE_SUPPORTS(sse2_), raw_type, wide<T, N> a0) noexcept
-    requires std::same_as<abi_t<T, N>, x86_128_>
+  EVE_FORCEINLINE wide<T, N> rec_(EVE_SUPPORTS(sse2_), raw_type, wide<T, N> const &v) noexcept
+      requires x86_abi<abi_t<T, N>>
   {
-    if constexpr(std::is_same_v<T, double>)
+    constexpr auto c = categorize<wide<T, N>>();
+
+          if constexpr(c == category::float32x16 )  return _mm512_rcp14_ps(v);
+    else  if constexpr(c == category::float64x8  )  return _mm512_rcp14_pd(v);
+    else  if constexpr(c == category::float32x8  )
+    {
+      if constexpr(current_api >= avx2 )            return _mm256_rcp14_epi32(v);
+      else                                          return _mm256_rcp_ps(v);
+    }
+    else  if constexpr(c == category::int64x4 )
     {
-      // The error for this approximation is no more than 1.5.e-12
-      return _mm_cvtps_pd(_mm_rcp_ps(_mm_cvtpd_ps(a0)));
+      if constexpr(current_api >= avx2 )            return _mm256_rcp14_epi64(v);
```

Comment:
This function doesn't exist.
The whole rcp is only available on real so why bother with integer ?

Reply:
mistakes corrected